src/commands/generateApiFromCode.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import * as fs from "fs";
+import * as vscode from 'vscode';
+import { ExceedTokenLimit, MODEL_SELECTOR } from "../constants";
+import { UiStrings } from "../uiStrings";
+import { GeneralUtils } from "../utils/generalUtils";
+
+export namespace GenerateApiFromCode {
+    export async function generateApiFromCode(context: IActionContext) {
+        const activeEditor = vscode.window.activeTextEditor;
+
+        if (!activeEditor) {
+            throw new Error(UiStrings.NoActiveFileOpen);
+        }
+
+        const languageId = activeEditor.document.languageId;
+        const fileUri = activeEditor.document.uri;
+
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: UiStrings.GeneratingOpenAPI
+        }, async (progress, token) => {
+            try {
+                const codeContent = await fs.promises.readFile(fileUri.fsPath, { encoding: 'utf-8' });
+
+                const messages = [
+                    vscode.LanguageModelChatMessage.User(`You are an expert in ${languageId} programming language and OpenAPI.
+Generate the OpenAPI Specification from the provided ${languageId} programming language.
+Try your best to parse the code and understand the code structure.
+Only return the specification content with YAML format, without any additional information.
+If the code is not REST API related, return "Not REST API related code" and provide an explanation.
+If this task can't be completed, return "Sorry, I can't assist" and provide an explanation.
+Here's the ${languageId} code of Web API:
+\`\`\`
+${codeContent}
+\`\`\``),
+                ];
+
+                let llmResponseText = '';
+                for (let i = 0; i < 5; i++) {
+                    const [model] = await vscode.lm.selectChatModels(MODEL_SELECTOR);
+                    const llmResponse = await model.sendRequest(messages, {}, new vscode.CancellationTokenSource().token);
+                    llmResponseText = '';
+                    for await (const fragment of llmResponse.text) {
+                        llmResponseText += fragment;
+                    }
+                    if (!(llmResponseText.toLocaleLowerCase().includes('sorry') && (llmResponseText.includes('can\'t') || llmResponseText.includes('cannot')))) {

```

Comment:
maybe these keyword can be turn into an negativity array for check.

Reply:
It is `and` + `or` logic, not a simple negativity array.